### PR TITLE
Feat: add GUI basic implementation (Texts, Component interface, Abstract Container)

### PR DESCRIPTION
**Pull request by MAILLOTNathan**

### CODE REVIEW

File: client/include/Engine.hpp
Comment (ju-lit-hein) [inline client/include/Engine.hpp:1]: @MAILLOTNathan 
Why are you including "GUI/Text.hpp" here ?

**Review by ju-lit-hein — CHANGES_REQUESTED** [review]

File: client/include/Engine.hpp
Comment (MAILLOTNathan) [reply client/include/Engine.hpp:1]: It's fixed

**Comment by MAILLOTNathan** [comment]
@ju-lit-hein

**Comment by MAILLOTNathan** [comment]
@ju-lit-hein

**Review by ju-lit-hein — APPROVED** [review]

**Pull request merged**